sds/src/proximity_keywords/mod.rs

```diff
@@ -12,191 +18,128 @@ use regex_syntax::ast::{
+    {
+        return Err(TooManyKeywords);
+    }
+
+    if let Some((content_regex, path_regex)) = compile_keywords(
+        config.included_keywords.clone(),
+        config.look_ahead_character_count,
+        &[],
+    )? {
+        included_keywords = Some(CompiledIncludedProximityKeywords {
+            look_ahead_character_count: config.look_ahead_character_count,
+            keywords_pattern: ProximityKeywordsRegex {
+                content_regex,
+                path_regex,
+            },
         });
+    }
 
-        Ok(CompiledProximityKeywords {
+    if let Some((content_regex, path_regex)) = compile_keywords(
+        config.excluded_keywords.clone(),
+        config.look_ahead_character_count,
+        EXCLUDED_KEYWORDS_REMOVED_CHARS,
+    )? {
+        excluded_keywords = Some(CompiledExcludedProximityKeywords {
             look_ahead_character_count: config.look_ahead_character_count,
-            included_keywords_pattern: included_patterns,
-            excluded_keywords_pattern: excluded_patterns,
-            metrics: Metrics::new(labels),
-        })
+            keywords_pattern: ProximityKeywordsRegex {
+                content_regex,
+                path_regex,
+            },
+            false_positive_counter: counter!(
+                "false_positive.proximity_keywords",
+                labels.clone_with_labels(Labels::new(&[(TYPE, "excluded_keywords".to_string())]))
+            ),
+        });
     }
 
-    fn contains_keyword_in_path(&self, path: &str, regex: &ProximityKeywordsRegex<false>) -> bool {
-        let input = Input::new(path).earliest(true);
+    Ok((included_keywords, excluded_keywords))
+}
 
-        regex.path_regex.search_half(&input).is_some()
-    }
+pub fn contains_keyword_in_path(path: &str, regex: &ProximityKeywordsRegex) -> bool {
+    let input = Input::new(path).earliest(true);
+    regex.path_regex.search_half(&input).is_some()
+}
 
-    /// Returns the match context which is what is searched for keywords
-    /// and the range where matches are searched for. The range is needed since the context is
-    /// expanded to ensure regex assertions (e.g. word boundaries) work correctly.
-    fn contains_keyword_match<const EXCLUDED_CHARS: bool>(
-        &self,
-        content: &str,
-        match_start: usize,
-        look_ahead_char_count: usize,
-        regex: &ProximityKeywordsRegex<EXCLUDED_CHARS>,
-    ) -> bool {
-        if EXCLUDED_CHARS {
-            let prefix_start_info = get_prefix_start(
-                match_start,
-                // Adding 1 to the start to account for assertion checking
-                look_ahead_char_count + 1,
-                content,
-            );
+/// Returns the match context which is what is searched for keywords
+/// and the range where matches are searched for. The range is needed since the context is
+/// expanded to ensure regex assertions (e.g. word boundaries) work correctly.
+fn contains_keyword_match(
```

Comment:
This method is now only used in excluded keywords. It's also specific to them due to the `EXCLUDED_KEYWORDS_REMOVED_CHARS` replacement line 127. Would it make sense to move it in `excluded_keywords.rs`?

Reply:
https://github.com/DataDog/dd-sensitive-data-scanner/pull/68/commits/8be9f95221846052b7352169b7d53a24e9375e90